pkg/operator/ceph/object/rgw-readiness-probe.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+set -x
+
+# For the readiness probes, the RGW should be removed from the load balancer (service) pool if
+# clients can't reach it to eliminate unnecessary client I/O failures.
+
+RGW_URL="0.0.0.0:8080"
+SWIFT_HEALTHCHECK_URL="$RGW_URL/swift/healthcheck"
+
+function check() {
+  local URL="$1"
+  # --silent - don't output progress info
+  # --output /dev/stderr - output HTML header to stdout (good for debugging)
+  # --write-out '%{response_code}' - print the HTTP response code to stdout
+  curl --silent --output /dev/stderr --write-out '%{response_code}' "$URL"
+}
+
+RATE_LIMITED_HTTP_CODE=498 # 498 for swift
+http_response="$(check $SWIFT_HEALTHCHECK_URL)"
+ec=$?
+
+if [[ $ec -ne 0 ]]; then
```

Comment:
If for any failure we are falling back to another check, why even bother with the swift check? As is, it seems like only the fallback call will result in ultimately determining the successful or failed probe.

Reply:
I kind of wonder that myself. Maybe we should just do the s3 check, but I don't know what the swift healthcheck does. It's not documented what expectations users can have of it. I just have the guidance from @cbodley that the healthcheck endpoint is preferred.

Reply:
i recommended the swift healthcheck API because it's specifically for this purpose. the s3 bucket listing API has similar properties (no rados i/o) only in the special case of unauthenticated requests. why is it necessary to have a fallback?

Reply:
I'm not sure if it's necessary to have a fallback or not. To my logic, if we try to curl the RGW without the swift path, we can be more certain that there is an issue connecting to the RGW. There are user config options that can change the swift path, which means there is a way in which user action can cause the health check to fail when the RGW is still accessible. Maybe it's best to just use the `curl 0.0.0.0:8080` check.

Reply:
> There are user config options that can change the swift path

hey @BlaineEXE, you're right that there's a https://docs.ceph.com/en/latest/radosgw/config-ref/#confval-rgw_swift_url_prefix that controls the path of this `/swift/healthcheck` endpoint. the reason that option exists is to support hosting swift at the root with s3 disabled (see the `Warning` about that in the docs)

in that configuration, your `curl 0.0.0.0:8080` command fails with `HTTP/1.1 405 Method Not Allowed` because swift hosts its account APIs at that / endpoint, and you have to authenticate to use them

it's probably not worth trying to support that configuration in rook, especially if you expect higher-level features like multisite replication and object bucket claims to work